contracts/DAORegistryScheme.sol

```diff
@@ -0,0 +1,192 @@
+    }
+
+    mapping(bytes32=>Parameters) public parameters;
+
+    /**
+     * @dev execution of proposals, can only be called by the voting machine in which the vote is held.
+     * @param _proposalId the ID of the voting in the voting machine
+     * @param _param a parameter of the voting result, 1 yes and 2 is no.
+     */
+    function executeProposal(bytes32 _proposalId, int256 _param) external onlyVotingMachine(_proposalId) returns(bool) {
+        Avatar avatar = proposalsInfo[msg.sender][_proposalId].avatar;
+        DAORegistryProposal memory proposal = organizationsProposals[address(avatar)][_proposalId];
+
+        if (_param == 1) {
+            Parameters memory params = parameters[getParametersFromController(avatar)];
+            bytes memory genericCallReturnValue;
+            bool success;
+            ControllerInterface controller = ControllerInterface(avatar.owner());
+            (success, genericCallReturnValue) =
+                controller.genericCall(params.contractToCall, proposal.callData, avatar, proposal.value);
+            if (success) {
+                delete organizationsProposals[address(avatar)][_proposalId];
+                emit ProposalDeleted(address(avatar), _proposalId);
+                emit ProposalExecuted(address(avatar), _proposalId, _param, genericCallReturnValue);
+            } else {
+                require(success, "proposal external call cannot be executed");
+            }
+        }
+
+        return true;
+    }
+
+    /**
+    * @dev hash the parameters, save them if necessary, and return the hash value
+    */
+    function setParameters(
+        bytes32 _voteParams,
+        IntVoteInterface _intVote,
+        address _contractToCall
+    ) public returns(bytes32)
+    {
+        bytes32 paramsHash = getParametersHash(_voteParams, _intVote, _contractToCall);
+        parameters[paramsHash].voteParams = _voteParams;
+        parameters[paramsHash].intVote = _intVote;
+        parameters[paramsHash].contractToCall = _contractToCall;
+        return paramsHash;
+    }
+
+    function getParametersHash(
+        bytes32 _voteParams,
+        IntVoteInterface _intVote,
+        address _contractToCall
+    ) public pure returns(bytes32)
+    {
+        return keccak256(abi.encodePacked(_voteParams, _intVote, _contractToCall));
+    }
+
+    /**
+    * @dev create a proposal to register a DAO with to a named registry
+    * @param _avatar the address of the organization the resource will be registered for
+    * @param _registryName the name of the registry to add the resource
+    * @param _proposedAvatar the organization we want to add to the registry
+    * @param _callData the abi call to add _proposedAvatar to the registry
+    * @param _value value(ETH) to transfer with the call
+    * @return a proposal Id
+    */
+    function proposeToAddDAO(
+        Avatar _avatar,
+        string memory _registryName,
+        Avatar _proposedAvatar,
+        bytes memory _callData,
```

Comment:
how about passing the registered dao + name .
the encoded abi can be build inside the contract.
this will make the ui easier and nicer I guess.